src/sentry/auth/helper.py

```diff
@@ -98,10 +105,15 @@ class AuthIdentityHandler:
 
     auth_provider: AuthProvider
     provider: Provider
-    organization: Organization
+    organization: ApiOrganization
     request: HttpRequest
     identity: Mapping[str, Any]
 
+    def __post_init__(self) -> None:
+        # For debugging. TODO: Remove when tests are stable
+        if not isinstance(self.organization, ApiOrganization):
```

Comment:
👍 
You could also make this conditional to tests by 

```
import sys
if 'pytest' in sys.modules:
   ...
```

Which means you can safely continue to leave it.